lib/SparqlGenerator.js

```diff
@@ -257,7 +303,9 @@ function toUpdate(update) {
 function isString(object) { return typeof object === 'string'; }
 
 // Maps the array with the given function, and joins the results using the separator
-function mapJoin(array, func, sep) { return array.map(func).join(isString(sep) ? sep : ' '); }
+Generator.prototype.mapJoin = function (array, func, sep) {
```

Comment:
Other argument seems fine. Perhaps the argument order should be `array, sep, func, self`.